### PR TITLE
Support Fix All in Solution for Enable Nullable Reference Types refactoring

### LINKED ISSUE #65852: Automating nullable enable over thousands of csproj
**Brief description:**

I have a couple of thousand csproj and millions of lines of code in our repository. I'd like to start the journey of moving to `#nullable enable` everywhere, but the current analyzers require that I insert `#nullable restore` into a file and perform operations through the VS IDE. I'm looking for a way to make large changes by invoking analyzers from the command line. I've done this in the past with dotnet-format.

To start, I'd like to invoke these two:

"Enable Nullable reference types in project"
"Remove Redundant Nullable directive"

**Languages applicable:**

C# only

@sharwell @mjsabby

### PR DESCRIPTION
**Pull request by sharwell**
Fixes #65852

### CODE REVIEW

File: src/Features/CSharp/Portable/CodeRefactorings/EnableNullable/EnableNullableCodeRefactoringProvider.cs [hunk 1]
```diff
@@ -62,6 +61,27 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
...
+
+        private static async Task<Solution> EnableNullableReferenceTypesAsync(Solution solution, CodeActionPurpose purpose, CodeActionOptionsProvider fallbackOptions, CancellationToken cancellationToken)
+        {
+            foreach (var projectId in solution.ProjectIds)
+            {
+                var project = solution.GetRequiredProject(projectId);
+                if (!CanEnableNullableInProject(project))
+                    continue;
+
+                solution = await EnableNullableReferenceTypesAsync(project, purpose, fallbackOptions, cancellationToken).ConfigureAwait(false);
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/CodeRefactorings/EnableNullable/EnableNullableCodeRefactoringProvider.cs:79]: nit.  consider doing all projects in parallel and then merging the result at the end.  this seems like a really good case to get multi-core scaling on.

**Comment by mavasani** [comment]
I have https://github.com/dotnet/roslyn/pull/61879 which adds this same FixAll support. The PR also adds unit tests. Feel free to pick up the tests and/or any logic from that PR if you'd like.

File: src/Features/CSharp/Portable/CodeRefactorings/EnableNullable/EnableNullableCodeRefactoringProvider.cs [hunk 2]
```diff
@@ -252,6 +272,33 @@ private enum CodeActionPurpose
             Apply,
         }
 
+        private sealed class CustomFixAllProvider : FixAllProvider
+        {
+            public static readonly CustomFixAllProvider Instance = new();
+
+            public override IEnumerable<FixAllScope> GetSupportedFixAllScopes()
+            {
+                yield return FixAllScope.Project;
```
Comment (mavasani) [inline src/Features/CSharp/Portable/CodeRefactorings/EnableNullable/EnableNullableCodeRefactoringProvider.cs:281]: I don't think FixAll in project makes sense here - that is the default behavior for this refactoring.

**Comment by sharwell** [comment]
Duplicate of #61879

**Pull request closed without merging**